enterprise/views.py

```diff
@@ -0,0 +1,184 @@
+"""
+User-facing views for the Enterprise app.
+"""
+from __future__ import absolute_import, unicode_literals
+
+import os
+
+from django.conf import settings
+from django.core.urlresolvers import reverse
+from django.http import Http404
+from django.shortcuts import redirect
+from django.utils.translation import ugettext as _
+from django.views.generic import View
+
+try:
+    from edxmako.shortcuts import render_to_response
+except ImportError:
+    render_to_response = None
+
+try:
+    from edxmako.paths import add_lookup
+except ImportError:
+    add_lookup = None
+
+try:
+    from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+except ImportError:
+    configuration_helpers = None
+
+try:
+    from third_party_auth.pipeline import get_complete_url, get_real_social_auth_object
+except ImportError:
+    get_complete_url = None
+    get_real_social_auth_object = None
+
+from enterprise.models import EnterpriseCustomer, EnterpriseCustomerUser, UserDataSharingConsentAudit
+from enterprise.tpa_pipeline import active_provider_enforces_data_sharing, get_enterprise_customer_for_request
+from enterprise.utils import NotConnectedToEdX
+
+
+class GrantDataSharingPermissions(View):
+    """
+    Provide a form and form handler for data sharing consent.
+
+    View handles the case in which we get to the "verify consent" step, but consent
+    hasn't yet been provided - this view contains a GET view that provides a form for
+    consent to be provided, and a POST view that consumes said form.
+    """
+
+    @staticmethod
+    def patch_mako_lookup():
+        """
+        Update the EdX Mako paths to point to our consent template.
+        """
+        if add_lookup is None:
+            raise NotConnectedToEdX(
+                _('Must be installed in an OpenEdX environment to render this response.')
+            )
+        full_location = os.path.realpath(__file__)
+        directory = os.path.dirname(full_location)
+        template_location = os.path.join(directory, 'templates', 'enterprise')
+        add_lookup('main', template_location)
```

Comment:
This code should be moved to the `enterprise` app's `AppConfig.ready()` section. Doing it on every request is redundant and will leak memory. Or am I missing something that makes this only have effect for one request-response cycle?